hiperwalk/plot/_plot.py

```diff
@@ -258,12 +258,9 @@ def plot_probability_distribution(
         # saves or shows image (or both)
         if not animate:
             if filename is not None:
-                # enumarating the plot
-                filename_suffix = (
-                    '-' + (len(probabilities)-1)//10 * '0' + str(i)
-                    if len(probabilities) > 1 else ''
-                )
-                plt.savefig(filename + filename_suffix)
+                # TODO: consider using Python's string formatting operations
+                filename_suffix = str(i).zfill(len(str(len(probabilities))))
```

Comment:
Since we start the count from zero,  `len(probabilities) - 1` should be used. Otherwise instead of creating files `fig-0.png`, ..., `fig-9.png`, the files `fig-00.png`, ..., `fig-09.png` are created.

Reply:
[abf6e53](https://github.com/hiperwalk/hiperwalk/pull/10/commits/abf6e532b0cf27d48d3eee9cf3cedfc36a714f73) fixes this.